codec/src/main/java/io/netty/handler/codec/DelimitedLengthFieldBasedFrameDecoder.java

```diff
@@ -0,0 +1,217 @@
+                lengthStr
+            ),
+            e
+        );
+      }
+
+      if (frameLength < 0) {
+        throw new CorruptedFrameException("negative pre-adjustment length field: " + frameLength);
+      }
+
+      frameLength += lengthAdjustment;
+
+      //consume length field and delimiter bytes
+      in.skipBytes(delimIndex + delimiter.capacity());
+
+      //consume delimiter bytes
+      consumingLength = false;
+    }
+
+    if (frameLength > maxFrameLength) {
+      long discard = frameLength - in.readableBytes();
+      tooLongFrameLength = frameLength;
+
+      if (discard < 0) {
+        // buffer contains more bytes then the frameLength so we can discard all now
+        in.skipBytes((int) frameLength);
+      } else {
+        // Enter the discard mode and discard everything received so far.
+        discardingTooLongFrame = true;
+        bytesToDiscard = discard;
+        in.skipBytes(in.readableBytes());
+      }
+      failIfNecessary(true);
+      return null;
+    }
+
+    // never overflows because it's less than maxFrameLength
+    int frameLengthInt = (int) frameLength;
+    if (in.readableBytes() < frameLengthInt) {
+      // need more bytes available to read actual frame
+      return null;
+    }
+
+    // the frame is now entirely present, reset state vars
+    consumingLength = true;
+    frameLength = 0;
+
+    // extract frame
+    int readerIndex = in.readerIndex();
+    int actualFrameLength = frameLengthInt;
+    ByteBuf frame = extractFrame(ctx, in, readerIndex, actualFrameLength);
+    in.readerIndex(readerIndex + actualFrameLength);
+    return frame;
+  }
+
+  private void failIfNecessary(boolean firstDetectionOfTooLongFrame) {
+    if (bytesToDiscard == 0) {
+      // Reset to the initial state and tell the handlers that
+      // the frame was too large.
+      long tooLongFrameLength = this.tooLongFrameLength;
+      this.tooLongFrameLength = 0;
+      discardingTooLongFrame = false;
+      consumingLength = false;
+      this.frameLength = 0;
+      if (!failFast || failFast && firstDetectionOfTooLongFrame) {
+        fail(tooLongFrameLength);
+      }
+    } else {
+      // Keep discarding and notify handlers if necessary.
+      if (failFast && firstDetectionOfTooLongFrame) {
+        fail(tooLongFrameLength);
+      }
+    }
+  }
+
+  /**
+   * Extract the sub-region of the specified buffer.
+   * <p>
+   * If you are sure that the frame and its content are not accessed after
+   * the current {@link #decode(ChannelHandlerContext, ByteBuf)}
+   * call returns, you can even avoid memory copy by returning the sliced
+   * sub-region (i.e. <tt>return buffer.slice(index, length)</tt>).
+   * It's often useful when you convert the extracted frame into an object.
+   * Refer to the source code of {@link ObjectDecoder} to see how this method
+   * is overridden to avoid memory copy.
+   */
+  protected ByteBuf extractFrame(ChannelHandlerContext ctx, ByteBuf buffer, int index, int length) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "ctx". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1172)